### PR TITLE
Align bookmarks search design to address bar design

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/414730916066338/1200979549371790/f

### Description
This PR includes the following:
- Create a basic standalone SearchBar component that is meant to be shared between pages (not yet included in this PR)
- Integrate the SearchBar with the Bookmarks Activity replacing the Android SearchView widget

### Steps to test this PR

Scenario 1: User has no bookmarks
- Open bookmarks page
- Search icon should not be visible

Scenario 2: User has bookmarks
- Open bookmarks page
- Click on the Search icon
- The search bar should appear and it should look like the address bar in the browse page
- Typing on the search bar should filter results as the user types
- Pressing back clears and hides the search bar
- Pressing clear clears the search bar

### UI changes
| Before  | After |
| ------ | ----- |
|before_bookmark_search|bookmark_search_clear|
|before_bookmark_search_result|bookmark_search_result|
|before_bookmark_empty|bookmark_search_empty|

### CODE REVIEW

**Review by joshliebe — COMMENTED** [review]
Looks good so far 🙂

I gave it a try and the filtering isn't working like it did before though, it should search all bookmarks, even the ones in folders (So when the search is empty it should show all folders and bookmarks in the list).

Had a look at the `SearchBar`, I'm happy with the approach you're taking with the widget 👍

File: app/src/main/java/com/duckduckgo/app/bookmarks/ui/SearchBar.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+class SearchBarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr), SearchBar {
+    private val binding: ComponentSearchBarBinding by viewBinding()
+
+    init {
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply app/src/main/java/com/duckduckgo/app/bookmarks/ui/SearchBar.kt]: Ah no real benefit actually. I changed this part to be done through xml and styles.

**Comment by karlenDimla** [comment]
I've fixed the incorrect behavior and moved the component to the common-ui module. I talked more to Sveta about this and we will get more feedback around this behavior before merging any changes. Just an FYI.

File: app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt [hunk 2]
```diff
@@ -249,23 +255,43 @@ class BookmarksActivity : DuckDuckGoActivity() {
...
+    override fun onBackPressed() {
+        if (searchBar.isVisible) {
+            hideSearchBar()
+        } else {
+            super.onBackPressed()
+        }
+    }
+
+    private fun showSearchBar() {
+        toolbar.visibility = GONE
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt:286]:
```suggestion
        toolbar.gone()
```

File: app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt [hunk 3]
```diff
@@ -249,23 +255,43 @@ class BookmarksActivity : DuckDuckGoActivity() {
...
+        viewModel.fetchBookmarksAndFolders()
+        searchBar.handle(SearchBar.Event.ShowSearchBar)
+    }
+
+    private fun hideSearchBar() {
+        toolbar.visibility = VISIBLE
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt:292]:
```suggestion
        toolbar.show()
```

File: app/src/main/res/layout/activity_bookmarks.xml [hunk 4]
```diff
@@ -17,16 +17,44 @@
...
 
-    <include
-        android:id="@+id/includeToolbar"
-        layout="@layout/include_default_toolbar" />
+    <com.google.android.material.appbar.AppBarLayout
+        android:id="@+id/appBarLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?toolbarBgColor"
+        android:theme="@style/ToolbarTheme">
```
Comment (malmstein) [inline app/src/main/res/layout/activity_bookmarks.xml:33]:
```suggestion
        android:theme="@style/Widget.DuckDuckGo.ToolbarTheme">
```

File: app/src/main/res/layout/activity_bookmarks.xml [hunk 5]
```diff
@@ -17,16 +17,44 @@
...
     tools:context="com.duckduckgo.app.bookmarks.ui.BookmarksActivity">
 
-    <include
-        android:id="@+id/includeToolbar"
-        layout="@layout/include_default_toolbar" />
+    <com.google.android.material.appbar.AppBarLayout
+        android:id="@+id/appBarLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?toolbarBgColor"
```
Comment (malmstein) [inline app/src/main/res/layout/activity_bookmarks.xml:32]: with the theme suggestion you can get rid of this

File: app/src/main/res/layout/activity_bookmarks.xml [hunk 6]
```diff
@@ -17,16 +17,44 @@
...
+        <FrameLayout
+            android:layout_width="match_parent"
+            android:layout_height="?attr/actionBarSize">
+
+            <androidx.appcompat.widget.Toolbar
+                android:id="@+id/toolbar"
+                android:layout_width="match_parent"
+                android:layout_height="?attr/actionBarSize"
+                android:background="?attr/toolbarBgColor"
+                android:theme="@style/ToolbarTheme"
```
Comment (malmstein) [inline app/src/main/res/layout/activity_bookmarks.xml:44]:
```suggestion
                android:theme="@style/Widget.DuckDuckGo.ToolbarTheme"
```

File: app/src/main/res/layout/activity_bookmarks.xml [hunk 7]
```diff
@@ -17,16 +17,44 @@
...
+                app:popupTheme="@style/PopUpOverflowMenu" />
```
Comment (malmstein) [inline app/src/main/res/layout/activity_bookmarks.xml:45]:
```suggestion
                app:popupTheme="@style/Widget.DuckDuckGo.PopUpOverflowMenu" />
```

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt [hunk 8]
```diff
@@ -82,6 +82,7 @@ class SearchBarView @JvmOverloads constructor(
             }
             SearchBar.Event.ShowSearchBar -> {
                 clearText()
+                binding.omnibarTextInput.showKeyboard()
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt:85]: this already calls `requestFocus()`, so you can get rid of the next line

**Review by malmstein — APPROVED** [review]
Couple of nits and suggestions, but this looks really good. Great work @karlenDimla!

**Pull request merged**